exercises/rectangles/src/example/java/RectangleCounter.java

```diff
@@ -15,13 +15,13 @@ int countRectangles(final String[] rawGrid) {
 
     private static final class Grid {
 
-        private enum Entry {
+        private enum Element {
             CORNER,
             HLINE,
             VLINE,
             SPACE;
 
-            private static Entry fromChar(final char rawGridEntry) {
+            private static Element fromChar(final char rawGridEntry) {
```

Comment:
I sense an intent to want to pair-up the input and the output, by picking a similar name; if so, that got lost in the rename of the output type.

----

tl;dr — `Piece`?

I couldn't help myself... I had to attempt to suggest names for these little bits of the diagram (not that `Element` isn't "right"... just nice having options).  In that journey I discovered that the Dictionary/Thesaurus that comes OOTB with OS X has a neat little section on some words called "Choose the right word".  There's one such entry for "part".  It reads like this:

> The whole is equal to the sum of its parts — **part** being a general term for any of the components of a whole. But how did the whole come apart? **Fragment** suggests that breakage has occurred (fragments of the vase) and often refers to a brittle substance such as glass or pottery. **Segment** suggests that the whole has been separated along natural or pre-existing lines of division (a segment of an orange), and **section** suggests a substantial and clearly separate part that fits closely with other parts to form the whole (a section of the bookcase). **Fraction** usually suggests a less substantial but still clearly delineated part (a fraction of her income), and a **portion** is a part that has been allotted or assigned to someone (her portion of the program). Finally, the very frequently used **piece** is any part that is separate from the whole (we're missing two pieces from the jigsaw puzzle).